control-plane/api-gateway/controllers/finalizer_test.go

```diff
@@ -0,0 +1,77 @@
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestEnsureFinalizer(t *testing.T) {
```

Comment:
do we want to throw a `t.Parallel()` on this and the other test in this file?

Reply:
Can't hurt!